js/libs/core.js

```diff
@@ -67,6 +67,60 @@ Fliplet.FormBuilder = (function() {
         component.methods = {};
       }
 
+      component.methods.setSourceValue = function(data) {
+        var result;
+        var $vm = this;
+
+        switch (data.source) {
+          case 'profile':
+            if (!data.key) {
+              throw new Error('A key is required to fetch data from the user\'s profile');
+            }
+
+            result = Fliplet.Profile.get(data.key);
```

Comment:
User profile data should be loaded from `Fliplet.User.getCachedSession()`, not `Fliplet.Profile.get()`.

Once the session is retrieved, look for the profile key from the `dataSource` object in `sessions`. If it doesn't exist, look for it in the `saml2` object.

Reply:
@tonytlwu 
Fliplet.Profile.get() - was taken from [here](https://github.com/Fliplet/fliplet-widget-form-builder/blob/f0f0fedf0792c810f7d77345068ae7d4c4a9e428/js/libs/form.js#L812)
Should we change it as well?

Reply:
@squallstar What are your thoughts on this, taking https://github.com/Fliplet/fliplet-studio/issues/5171 into consideration.

I imagine we don't want to change how field.set() works to avoid breaking changes, but I also don't want to start using `Fliplet.Profile.get()` for this new feature because I want to be able to load any user data from the session entries.

Reply:
@tonytlwu should we have fallbacks and load the first value we find for the key in these namespaces?

1. Session - dataSource passport
2. Session - saml2 passport
3. Session - any passport 
4. Profile

If the logic is clear enough in the future we can potentially port it to a new core JS API as we discussed in the ticket.

Reply:
@squallstar That works for me.

@YaroslavOvdii See above. And be sure to resolve conflicts before it's ready for review.

Reply:
@squallstar @tonytlwu 
Do I get It right that we should look for the key as I show below?

![image](https://user-images.githubusercontent.com/53430352/90020088-07224100-dcb8-11ea-8800-54820dff8690.png)

Reply:
@YaroslavOvdii nope, you need to look into `session.entries`, see https://developers.fliplet.com/API/fliplet-session.html#read-details-about-connected-accounts

Reply:
@squallstar I get it now, thank you.